docs/src/charm/howto/custom-workers.md

```diff
@@ -0,0 +1,62 @@
+## Adding Worker Nodes with Custom Configurations
+
+This guide will walk you through how to deploy multiple `k8s-worker`
+applications with different configurations, to create node groups with specific
+capabilities or requirements.
+
+## What you'll need
+
+This guide assumes the following:
+- A working Kubernetes cluster deployed with the `k8s` charm
+
+### Example Worker Configuration
+
+In this example, we will create two `k8s-worker` applications with different
+configuration options.
+
+```{note}
+The configurations shown below are examples to demonstrate the deployment
+pattern. You should adjust the node configurations, labels, and other
+parameters according to your specific infrastructure requirements, workload
+needs, and organizational policies. Review the [charm configuration] options
+documentation to understand all available parameters that can be customized for
+your worker nodes.
+```
+
+1. Workers for memory-intensive workloads (`worker-memory-config.yaml`):
+```yaml
+memory-workers:
+  bootstrap-node-taints: "workload=memory:NoSchedule"
+  kubelet-extra-args: "system-reserved=memory=2Gi"
+```
+
+2. Workers for GPU workloads (`worker-gpu-config.yaml`):
+```yaml
+gpu-workers:
+  bootstrap-node-taints: "accelerator=nvidia:NoSchedule"
+  node-labels: "gpu=true"
+```
+
+Deploy the worker applications with the custom configurations:
```

Comment:
We are deploying and integrating into our k8s cluster so I would also add this here